# Case Study on Testing of Hypothesis  

A company started to invest in digital marketing as a new way of their product promotions.For that they collected data and decided to carry out a study on it.

● The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.

● The company needs to check whether there is any dependency between the features “Region” and “Manager”.

   Help the company to carry out their study with the help of data provided.

In [1]:
#importing pandas,numply,matplotlib.pyplot,seaborn libaries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
#Reading the dataset to the python environment
data=pd.read_csv('Sales_add.csv')

In [3]:
#displaying  top 5  rows of the dataset
data.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

In [4]:
#To know data types of each columns and checking for null values in the columns
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1008.0+ bytes


Note-         
                                                                                                                       It appears that we do not have empty/NaN values in the dataset, as the number of Non-Null counts for each column is equal to 22, which is exactly the number of rows in the dataset. 
                                                                                                                       
We also note that our numeric data are detected as integer values.

In [5]:
#Check for the null values present in each column of  dataset using 'isnull().sum()' function
data.isnull().sum()

Month                             0
Region                            0
Manager                           0
Sales_before_digital_add(in $)    0
Sales_After_digital_add(in $)     0
dtype: int64

Note-

No Null values

In [6]:
# to know  statistical summary of integer datatype columns 
data.describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

In [7]:
#renaming columns
data=data.rename(columns={'Sales_before_digital_add(in $)':'Sales_before','Sales_After_digital_add(in $)':'Sales_after'})

In [8]:
#to check new name of the columns
data.head()

Month      Region      Manager  Sales_before  Sales_after
0  Month-1  Region - A  Manager - A        132921       270390
1  Month-2  Region - A  Manager - C        149559       223334
2  Month-3  Region - B  Manager - A        146278       244243
3  Month-4  Region - B  Manager - B        152167       231808
4  Month-5  Region - C  Manager - B        159525       258402

### ● The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.

>Set the hypothesis 

Null Hypothesis (H0): no increase in sales after stepping into digital marketing

Alternate Hypothesis (H1):  increase in sales after stepping into digital marketing.

Here we check

H0 < H1

Also sample is less than 30

 Test we use is - 
***one tailed Paired sampled t-test***




>Set the significance level 


Significance  level= 0.05

In [9]:
data['Sales_before'].mean()

149239.95454545456

In [10]:
data['Sales_after'].mean()

231123.72727272726

Find-

mean of sales after stepping into digital marketing is much higher than sales before stepping into digital marketing

>Computing the test statistics 

In [14]:
from scipy.stats import ttest_ind
import scipy.stats as stats

In [71]:
t_stat, p_val = stats.ttest_rel(data['Sales_after'], data['Sales_before'], alternative='greater')
print('t-value:',t_stat,'p-value:',p_val)

t-value: 12.09070525287017 p-value: 3.168333502287889e-11


In [72]:
print( "p value " , round(p_val,4))
if p_val > 0.05:
    print('Null Hypothesis is proved')
else :
    print ('Alternate Hypothesis is proved')

p value  0.0
Alternate Hypothesis is proved


>Making a Decision 

FInd-

*From the Ttest performed,*
        
pvalue=0.0 < 0.05 

so we  reject the null Hypothesis

<span style="color:red">there is increase in sales after stepping into digital marketing.</span>

---


###  The company needs to check whether there is any dependency between the features “Region” and “Manager”.

###### If we want to check the relationship between two categorical values, we will use the Chi-Squared test.


>Set the hypothesis

Null Hypothesis:no dependency between the features “Region” and “Manager”.

Alternate Hypothesis: there is dependency between the features “Region” and “Manager”

>Set the significance level

Significance level= 0.05


>Computing the test statistics

In [20]:
#The easiest way to apply a chi-squared test is to compute the contigency table.

from scipy.stats import chi2_contingency
contigency= pd.crosstab(data['Region'], data['Manager'])
contigency

Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1

Note-

     Contingency Table  is tally of counts between two or more categorical variables

     Contingency Tables are giving clear correlation values between two and more variables. Thus making it much more                      useful to understand the data for further information extraction.

In [62]:
#Chi-square test of independence.
Cal_chi, p, dof, expected = chi2_contingency(contigency)
print(f"Calculated Chi Square value:     {Cal_chi}")
print(f"p-value:            {round(p,2)}")
print(f"degrees of freedom: {dof}")
print("expected frequencies:")
print(expected)

Calculated Chi Square value:     3.050566893424036
p-value:            0.55
degrees of freedom: 4
expected frequencies:
[[4.09090909 3.18181818 2.72727273]
 [2.86363636 2.22727273 1.90909091]
 [2.04545455 1.59090909 1.36363636]]


In [35]:
import scipy.stats

#find Chi-Square critical value
chi_critical=scipy.stats.chi2.ppf(1-.05, df=4)
print(f"Chi-Square critical value:{chi_critical}")

Chi-Square critical value:9.487729036781154


>Making a Decision 

In [67]:
if((p >=0.05) and (chi_critical>Cal_chi)):
    print('Null Hypothesis is proved')
else :
    print ('Alternate Hypothesis is proved')

Null Hypothesis is proved


FInd-

From the Chi-square test performed,

    pvalue=0.55 > 0.05 
    Chi-Square critical value>Calculated Chi Square value
    so we  accept the null Hypothesis
   <span style="color:red">there is no dependency between the features “Region” and “Manager”</span>

Submitted by-

`Kevin M
 DSA Norka B3 2021`

---
___

***